sdk/eventhub/event-hubs/review/event-hubs.api.md

```diff
@@ -39,6 +39,21 @@ export interface Checkpoint {
     sequenceNumber: number;
 }
 
+// @public
+export class CheckpointManager {
+    // (undocumented)
+    updateCheckpoint(eventData: EventData): Promise<void>;
+    // (undocumented)
+    updateCheckpoint(offset: string, sequenceNumber: number): Promise<void>;
+}
+
+// @public
+export enum CloseReason {
```

Comment:
Is this something all languages are exposing? I don't remember seeing it in our API design, that `close()` takes a reason. 🤔

Reply:
[Proposed API for javascript](https://gist.github.com/ramya-rao-a/6ce030240112063e12a23477697b01f2), here it takes CloseReason.

Reply:
@conniey We brought that up on Monday too, because we were curious how Java would know why the subscription was closed.

Reply:
Yeah, I also wonder what the user could do differently with that reason (ie. user shutdown vs stolen)? I mean, we log it in the EventConsumer but it is not propagated out.

Reply:
In theory if it was a shutdown they could checkpoint one last time, whereas if it were stolen they wouldn’t be able to update the checkpoint.